pkg/network/protocols/http/monitor.go

```diff
@@ -149,6 +154,12 @@ func (m *Monitor) Start() error {
 		}
 	}()
 
+	if m.processMonitor != nil {
```

Comment:
what if `processMonitor == nil` ? you will return nil, instead of an error.

see my comment for GetProcessMonitor, I think we should bail out immediately with an error if we failed to initialize it inside GetProcessMonitor method